src/types/transaction.rs

```diff
@@ -799,7 +800,27 @@ pub struct V2FileContractResolution {
 }
 
 impl Encodable for V2FileContractResolution {
-    fn encode(&self, _encoder: &mut Encoder) { todo!() }
+    fn encode(&self, encoder: &mut Encoder) {
+        self.parent.encode(encoder);
+        // Determine resolution type from the resolution itself, matching Go implementation
+        match &self.resolution {
+            V2FileContractResolutionWrapper::Renewal(_) => {
+                encoder.write_u8(0);
+            },
+            V2FileContractResolutionWrapper::StorageProof(_) => {
+                encoder.write_u8(1);
+            },
+            V2FileContractResolutionWrapper::Expiration => {
+                encoder.write_u8(2);
+            },
+            V2FileContractResolutionWrapper::Finalization(_) => {
+                // Finalization is not in Go core, but we support it for compatibility
+                // If this is used, it should be encoded as type 3
+                encoder.write_u8(3);
+            },
```

Comment:
i believe we can remove all finalization variants. it doesn't seem to be a resolution type currently:

https://github.com/SiaFoundation/core/blob/b413f9dc2637f6428a957a9e4c27544e5e364a5d/types/types.go#L674-L681